### DISCUSSION TITLE
two questions about digital signatures

### DISCUSSION THREAD

**Discussion by ousia** [pymupdf/PyMuPDF#2544] [Q&A] [answered]
@JorjMcKie,

congratulations for keeping the excellent work developing these bindings.

I have two questions I couldn’t find anywhere else.

1. Are digital signatures implemented in the bindings? Is there an ETA for this feature?

    (Sorry, but signatures is a tricky string to search for in this repo, since there are CLAs.)

1. This issue happens also with MuPDF upstream, but I don’t know where to get info.

    Is there any way to make MuPDF handle security devices from Firefox (or to sign using a smart card)?

Many thanks for your help and congratulations again for your hard work.

**Comment by JorjMcKie** [comment, marked as answer]
Hi Pablo,

Ad 1.
No, not yet. Although support is present in MuPDF itself as you know, linking to OpenSSL, specifically automating OpenSSL access as a thirdparty library is a pain in the neck. Which aggravates with the many platforms PyMuPDF is supporting nowadays. It is a backburner however and hasn't gone from the enhancement list.

Ad 2.
This topic is new to me I must confess.
Are you a user of Artifex public Discord channels? Asking around there is probably the most appropriate place. Here are the links: [PyMuPDF](https://discord.com/channels/770681584617652264/983871937711341618), [MuPDF](https://discord.com/channels/770681584617652264/770685907279282196).

**Reply by ousia** [reply]
Hi Jorj,

many thanks for your fast reply.

I wonder whether these [Python bindings](https://cryptography.io/) for OpenSSL might help the integration.

Many thanks for your help again.
